### ISSUE TITLE
other shell support

### ISSUE THREAD

**Issue by VinSpee** [tmux-plugins/tmux-continuum#2]
great work, I'd love to see a version for fish Shell

**Comment by bruno-** [comment]
Hey,
from the top of my head - I'm not sure why you couldn't run this with fish shell? Are you getting any errors?
The scripts are written in bash, so as long you have that installed on your system the plugin should work even though you're using fish as your main shell in tmux.

I might be wrong with something here (I've never used fish shell), but a lot of zsh users are using tmux plugins and from what I know things work just fine.

**Comment by VinSpee** [comment]
"Auto-start" (with iterm) doesn't happen. No error, it just doesn't start on reboot. 



(null)

**Comment by bruno-** [comment]
Hmm.. maybe it's a bug. How about we investigate a bit more.. care to paste the content of your `tmux.conf`?

To confirm:
- auto save feature works?
- auto restore feature works? (if tmux is started, restore happens automatically?)

**Comment by VinSpee** [comment]
```sh
set -g default-terminal 'xterm-256color'

# Use vim keybindings in copy mode
setw -g mode-keys vi

# Set Titles
set-option -g set-titles on
set-option -g set-titles-string '[#S:#I #H] #W'

#Copy And Paste

# Setup 'v' to begin selection as in Vim
bind-key -t vi-copy v begin-selection
bind-key -t vi-copy y copy-pipe "reattach-to-user-namespace pbcopy"

# Update default binding of `Enter` to also use copy-pipe
#unbind -t vi-copy Enter
#bind-key -t vi-copy Enter copy-pipe "reattach-to-user-namespace pbcopy"

#Leave my window names alone!
setw -g automatic-rename off

# Use ctrl-a instead of ctrl-b
set -g prefix C-a
unbind C-b
bind C-a send-prefix

#Set window index to start at 1 instead of 0
set -g base-index 1

#Tile All Windows
unbind =
bind = select-layout tiled

# basic window operations
unbind c
bind c new-window
unbind &  # purposefully diffcult combination, better the shell's "C-d", or kill-pane
bind & confirm-before kill-window
unbind x  # close pane (if last also the window)
bind x confirm-before kill-pane

# add split side-by-side
unbind |
bind -r | split-window -h

# add split top-bottom
unbind _
bind -r _ split-window -v

# swap panes
unbind {
bind -r { swap-pane -U
unbind }
bind -r } swap-pane -D

# number panes visibly
unbind q
bind q display-panes

# cycle through pane layouts
unbind Space
bind -r Space next-layout

# Smart pane switching with awareness of vim splits
bind -n C-h run "(tmux display-message -p '#{pane_current_command}' | grep -iqE '(^|\/)vim$' && tmux send-keys C-h) || tmux select-pane -L"
bind -n C-j run "(tmux display-message -p '#{pane_current_command}' | grep -iqE '(^|\/)vim$' && tmux send-keys C-j) || tmux select-pane -D"
bind -n C-k run "(tmux display-message -p '#{pane_current_command}' | grep -iqE '(^|\/)vim$' && tmux send-keys C-k) || tmux select-pane -U"
bind -n C-l run "(tmux display-message -p '#{pane_current_command}' | grep -iqE '(^|\/)vim$' && tmux send-keys C-l) || tmux select-pane -R"
bind -n C-\ run "(tmux display-message -p '#{pane_current_command}' | grep -iqE '(^|\/)vim$' && tmux send-keys 'C-\\') || tmux select-pane -l"

# resize pane
unbind '('
bind -r '(' resize-pane -L 4
unbind ')'
bind -r ')' resize-pane -R 4
unbind +
bind -r + resize-pane -U 2
unbind -
bind -r - resize-pane -D 2


# command line
unbind :
bind : command-prompt


# display key bindings
unbind ?
bind ? list-keys

# Choose Session
bind S choose-session

# Activity
 setw -g monitor-activity on
 set -g visual-activity off

 # Better name management
 bind c new-window
 bind , command-prompt "rename-window '%%'"

 # Copy mode
 setw -g mode-keys vi
 bind ` copy-mode
 unbind [
 unbind p
 bind p paste-buffer
 bind -t vi-copy H start-of-line
 bind -t vi-copy L end-of-line
 bind -t vi-copy v begin-selection
 bind -t vi-copy Escape cancel

# Setup 'v' to begin selection as in Vim
#bind-key -t vi-copy y copy-pipe "reattach-to-user-namespace pbcopy"

# Update default binding of `Enter` to also use copy-pipe
#unbind -t vi-copy Enter
#bind-key -t vi-copy Enter copy-pipe "reattach-to-user-namespace pbcopy"

# Scroll your way into copy mode (scrollback buffer)
# and select text for copying with the mouse
set -g mode-mouse on
set -g mouse-resize-pane on
set -g mouse-select-pane on
set -g mouse-select-window on

# RESTORE NEOVIM SESSION
set -g @resurrect-strategy-nvim 'session'
set -g @continuum-restore 'on'
set -g @continuum-boot-options 'iterm'

source ~/.tmux.colors.conf

# List of plugins
# Supports `github_username/repo` or full git URLs
set -g @tpm_plugins "              \
  tmux-plugins/tpm                 \
  tmux-plugins/tmux-sensible       \
  tmux-plugins/tmux-resurrect      \
  tmux-plugins/tmux-continuum      \
  tmux-plugins/tmux-copycat        \
  tmux-plugins/tmux-yank           \
"

# Other examples:
# github_username/plugin_name    \
# git@github.com/user/plugin     \
# git@bitbucket.com/user/plugin  \

# initializes TMUX plugin manager
run-shell ~/.tmux/plugins/tpm/tpm

```

- Auto-save does not work. Manual save does.
- auto-restore does work

**Comment by bruno-** [comment]
Hey, thanks for the feedback.
I also experienced an issue with auto-save a couple days back. That should now be fixed, commit here: d6ac017d1dc7546ede2f66b2fb8015c720dce96a.

Can you please try updating a plugin and see if auto-save feature works then?

**Comment by bruno-** [comment]
Some users are having issues making the auto-restore feature work. There's issue #1 so let's discuss there.
